src/passes/OptimizeInstructions.cpp

```diff
@@ -350,6 +350,46 @@ struct OptimizeInstructions
           return inner;
         }
       }
+      {
+        // x <<>> (C & (31 | 63))   ==>   x <<>> C'
+        // x <<>> (y & (31 | 63))   ==>   x <<>> y
+        // where '<<>>':
+        //   '<<', '>>', '>>>'. 'rotl' or 'rotr'
+        Type type = curr->type;
+
+        if (type.isInteger()) {
+          BinaryOp op;
+          Const* c;
+          Expression *x, *y;
+
+          if (matches(curr, binary(&op, any(&x), constant(&c))) &&
+              Abstract::hasAnyShift(type, op)) {
+            // truncate RHS constant
+            // x <<>> (C & (31 | 63))   ==>   x <<>> C'
+            c->value = c->value.and_(
+              Literal::makeFromInt32(type.getByteSize() * 8 - 1, type));
```

Comment:
This is kind of confusing because the comment makes it look like you're matching a pattern with an `&`, but in fact the `&` is not actually in the pattern. Would you be able to write this in a way that doesn't modify `c->value`?